### ISSUE TITLE
Use shouldComponentUpdate to handle prop updates

### ISSUE THREAD

**Issue by theiliad** [carbon-design-system/carbon-charts#177]
Prop updates such as `data` should be handled this way, and update the chart data automatically...

**Comment by lordi** [comment]
Yea, react and d3 should be leveraged to have responsive, declarative charts.

Currently I'm using this workaround:

```
class ResponsiveLineChart extends LineChart {
 componentDidUpdate() {
   this.chart.setData(this.props.data);
   this.chart.update()
 }
}
```

It works, but does not update tooltip-points on the lines.

**Comment by theiliad** [comment]
This has been addressed in #275
